### PR TITLE
TChannelThrift UX improvements

### PR DESCRIPTION
**Pull request by Raynos**
In passing when trying to upgrade tchannel-gen to
tchannel@3 I want to make some user interface improvements

There are a bunch of changes in this PR and some are more
"debatable" then others.

I will break anything contraversial into a seperate PR.

r: @jcorbin @kriskowal

### CODE REVIEW

File: hyperbahn-client.js [hunk 1]
```diff
@@ -175,6 +176,26 @@ HyperbahnClient.prototype.setReportTracing = function setReportTracing(bool) {
     self.reportTracing = bool;
 };
 
+HyperbahnClient.prototype.getThriftCodecSync =
```
Comment (kriskowal) [inline hyperbahn-client.js]: Let’s drop Codec. Unnecessary jargon from nowhere.

Comment (Raynos) [inline hyperbahn-client.js]: `getThriftSync()` ?

**Comment by kriskowal** [comment]
Looks good to me. Could be getThriftSync or asThriftSync I thynk.

File: as/thrift.js [hunk 2]
```diff
@@ -66,17 +66,13 @@ function TChannelAsThrift(opts) {
         'channel must be provided with isHealthy');
 
     if (self.isHealthy) {
-        fs.readFile(path.join(__dirname, 'meta.thrift'), 'utf8', registerHealthCheck);
+        var thriftSource = fs.readFileSync(
```
Comment (jcorbin) [inline as/thrift.js]: sync io in constructor is sad, can we instead inject the content thru opts, and then provide a `TChannelThift.somethingSync` convenience?

File: as/thrift.js [hunk 3]
```diff
@@ -142,10 +138,29 @@ TChannelAsThrift.prototype.request = function request(reqOptions) {
...
+    return self.channel.waitForIdentified(options, cb);
+};
+
 TChannelAsThrift.prototype.register =
 function register(channel, name, opts, handle, spec) {
     var self = this;
 
+    // support register(endpoint, opts, handle)
+    if (typeof channel === 'string') {
+        assert(self.channel, 'channel is required for thrift.register()');
```
Comment (jcorbin) [inline as/thrift.js]: `assert(spec === undefined)` ?

File: as/thrift.js (hunk 2 above)
Comment (Raynos) [inline as/thrift.js]: Agreed it's bad. I wanted to fix the race condition. I can also fix race condition with `setTimeout` in tests.

The same race condition applies to anyone wanting to test their health check. Not sure if there is a good fix without breaking back compat.

File: hyperbahn-client.js (hunk 1 above)
Comment (jcorbin) [inline hyperbahn-client.js]: :+1: codec is 🙌weird🙌

File: as/thrift.js (hunk 3 above)
Comment (Raynos) [inline as/thrift.js]: good idea.

**Comment by malandrew** [comment]
`register()` in as-thrift seems a tad overloaded now. time to change the interface to an object with keys instead of an arity of 5? I know this would result in a major version bump, but it's something work considering.

**Comment by Raynos** [comment]
We've previously committed to arity of 5 as being acceptable.

File: hyperbahn-client.js (hunk 1 above)
Comment (malandrew) [inline hyperbahn-client.js]: Agree on codec being unnecessary jargon. `getThriftSync()`, `getThriftFile()`, `readThriftFile()`, etc. are all better. Also, get for me implies some sort of network fetching. For local disk i/o, `read` is preferable to `get` unless we plan to allow `options.thriftFile` to be something other than a local filepath.

**Comment by malandrew** [comment]
LGTM, however i think that the arity of 5 on register() only makes sense if the function signature is always the same with later args being optional. the fact that the first arg can be optional, makes me feel like we should use an object instead of arity of 5 and type sniffing. But again, that's a separate discussion.

**Pull request merged**